lang/src/OTS/IMP/org/partiql/ots/legacy/operators/LegacyBinaryConcatOp.kt

```diff
@@ -33,8 +33,15 @@ object LegacyBinaryConcatOp : ScalarOp {
         }
 
         // Here only VARCHAR or CHAR exists
-        val leftLength = lType.parameters[0]!!
-        val rightLength = rType.parameters[0]!!
+        fun checkUnlimitedVarchar(type: CompileTimeType) =
+            type.scalarType === VarcharType && type.parameters.isEmpty()
+
+        if (checkUnlimitedVarchar(lType) || checkUnlimitedVarchar(rType)) {
+            return Successful(StringType.compileTimeType)
+        }
+
+        val leftLength = lType.parameters.getOrElse(0) { 1 }
+        val rightLength = rType.parameters.getOrElse(0) { 1 }
```

Comment:
Let's not focus on code quality in each operator/function invoker, since anyway it will be refactored once we are working on supporting function overloading

Reply:
This comment applies to other operator/function invoker as well

Reply:
My bigger concern here that this might be introducing new logic that differs from what is on the main branch.

Reply:
I think there is no logic change here

Reply:
Line 43 & 44 specify the default type parameter of `CharType` to be 1, which is the same as `main` branch.

Reply:
The only difference is that in `main`, there is no unlimited `VarcharType`. Unlimited `VarcharType` there is considered as `StringType`.

Reply:
So something like `CAST(1 AS VARCHAR)` is actually `CAST(1 AS STRING)` in `main`.